x-pack/plugins/data_usage/server/routes/internal/data_streams_handler.ts

```diff
@@ -5,36 +5,50 @@
  * 2.0.
  */
 
-import { RequestHandler } from '@kbn/core/server';
+import { type ElasticsearchClient, RequestHandler } from '@kbn/core/server';
 import { DataUsageContext, DataUsageRequestHandlerContext } from '../../types';
-
 import { errorHandler } from '../error_handler';
 
+export interface MeteringStats {
+  name: string;
+  num_docs: number;
+  size_in_bytes: number;
+}
+
+interface MeteringStatsResponse {
+  datastreams: MeteringStats[];
+}
+
+const getMeteringStats = (client: ElasticsearchClient) => {
+  return client.transport.request<MeteringStatsResponse>({
+    method: 'GET',
+    path: '/_metering/stats',
+  });
+};
+
 export const getDataStreamsHandler = (
   dataUsageContext: DataUsageContext
 ): RequestHandler<never, unknown, DataUsageRequestHandlerContext> => {
   const logger = dataUsageContext.logFactory.get('dataStreamsRoute');
 
   return async (context, _, response) => {
-    logger.debug(`Retrieving user data streams`);
+    logger.debug('Retrieving user data streams');
 
     try {
       const core = await context.core;
-      const esClient = core.elasticsearch.client.asCurrentUser;
-
-      const { data_streams: dataStreamsResponse } = await esClient.indices.dataStreamsStats({
-        name: '*',
-        expand_wildcards: 'all',
-      });
+      const { datastreams: meteringStats } = await getMeteringStats(
+        core.elasticsearch.client.asSecondaryAuthUser
```

Comment:
@neptunian I copied some of this from [x-pack/plugins/index_management/server/routes/api/data_streams/register_get_route.ts#L90](https://github.com/elastic/kibana/blob/dffe0b571899b2ed0c71ee9f090095311d4d2b55/x-pack/plugins/index_management/server/routes/api/data_streams/register_get_route.ts#L90).  

Do you happen to know how `asSecondaryAuthUser` is different from `asCurrentUser`. I also tried with `asCurrentUser` here and it yields the same set of datastreams/data. I'm not sure if we should just use `asCurrentUser`.

Reply:
Good question.  Lets ask @elastic/kibana-management .

Reply:
Hey @mattkime! Do you or anyone on your team happen to know?

Reply:
I'm getting up to speed on this but to answer your direct question - https://github.com/elastic/kibana/blob/main/packages/core/elasticsearch/core-elasticsearch-server/src/client/scoped_cluster_client.ts#L28